experimental/packages/opentelemetry-instrumentation-fetch/src/fetch.ts

```diff
@@ -138,17 +166,28 @@ export class FetchInstrumentation extends InstrumentationBase<FetchInstrumentati
     response: FetchResponse
   ): void {
     const parsedUrl = web.parseUrl(response.url);
-    span.setAttribute(SEMATTRS_HTTP_STATUS_CODE, response.status);
-    if (response.statusText != null) {
-      span.setAttribute(AttributeNames.HTTP_STATUS_TEXT, response.statusText);
+
+    if (this._semconvStability & SemconvStability.OLD) {
+      span.setAttribute(ATTR_HTTP_STATUS_CODE, response.status);
+      if (response.statusText != null) {
+        span.setAttribute(AttributeNames.HTTP_STATUS_TEXT, response.statusText);
+      }
+      span.setAttribute(ATTR_HTTP_HOST, parsedUrl.host);
+      span.setAttribute(ATTR_HTTP_SCHEME, parsedUrl.protocol.replace(':', ''));
+      if (typeof navigator !== 'undefined') {
+        span.setAttribute(ATTR_HTTP_USER_AGENT, navigator.userAgent);
+      }
     }
-    span.setAttribute(SEMATTRS_HTTP_HOST, parsedUrl.host);
-    span.setAttribute(
-      SEMATTRS_HTTP_SCHEME,
-      parsedUrl.protocol.replace(':', '')
-    );
-    if (typeof navigator !== 'undefined') {
-      span.setAttribute(SEMATTRS_HTTP_USER_AGENT, navigator.userAgent);
+
+    if (this._semconvStability & SemconvStability.STABLE) {
```

Comment:
I think we are missing the case for DUPLICATE here, which I believe then matches up with a config change I commented on in one of the tests.

Reply:
The two if-blocks together handle the SemconvStability.DUPLICATE case. The enum values are intentionally setup so that DUPLICATE is `OLD | STABLE`.

```
% cat foo.ts
import {SemconvStability} from '@opentelemetry/instrumentation';

console.log('DUPLICATE:', SemconvStability.DUPLICATE);
console.log('OLD | STABLE:', SemconvStability.OLD | SemconvStability.STABLE);

const stab = SemconvStability.DUPLICATE;
if (stab | SemconvStability.OLD) {
  console.log('yes, emit old semconv')
}
if (stab | SemconvStability.STABLE) {
  console.log('yes, emit stable semconv')
}

% npx tsc --skipLibCheck foo.ts

% node foo.js
DUPLICATE: 3
OLD | STABLE: 3
yes, emit old semconv
yes, emit stable semconv
```

Reply:
stabby! You are right, there is no return in here to break out of the next if block.